src/interactive-window/editor-integration/types.ts

```diff
@@ -68,30 +78,35 @@ export interface ICellHash {
     timestamp: number;
     code: string; // Code that was actually hashed (might include breakpoint and other code)
     debuggerStartLine: number; // 1 based line in source .py that we start our file mapping from
+    startOffset: number;
+    endOffset: number;
+    deleted: boolean;
+    realCode: string;
+    trimmedRightCode: string;
+    firstNonBlankLineIndex: number; // zero based. First non blank line of the real code.
 }
 
 export interface IFileHashes {
     uri: Uri;
     hashes: ICellHash[];
 }
 
-export const ICellHashListener = Symbol('ICellHashListener');
-export interface ICellHashListener {
-    hashesUpdated(hashes: IFileHashes[]): Promise<void>;
+export const IGeneratedCodeStore = Symbol('IGeneratedCodeStore');
+export interface IGeneratedCodeStore {
+    clear(): void;
+    readonly all: IFileHashes[];
+    getCellHash(cell: NotebookCell): ICellHash | undefined;
+    getFileHashes(fileUri: Uri): ICellHash[];
```

Comment:
# Question: What do we call this method?

`getCellHash` is wrong again. Should it be `getForCell` and `getForFile` or `getCellInfo` and `getCellInfosByFile`, etc?

I want to get rid of this name `hash` from the method, it can be very confusing.
I'm not too fussed about the type `ICellHash` as that's rarely seen, its merely used to structure the data.

Reply:
getGeneratedCode?

Reply:
I liked code either for GeneratedCode or RuntimeCode. Seems like Generated code would work well. To me that expresses the intent / usage best.

Reply:
Done, thanks